source/_integrations/light.template.markdown

```diff
@@ -31,12 +31,17 @@ light:
         color_template: "({{states('input_number.h_input') | int}}, {{states('input_number.s_input') | int}})"
         turn_on:
           service: script.theater_lights_on
+          data: 
+            transition: "{{ transition | float}}"
         turn_off:
           service: script.theater_lights_off
+          data: 
+            transition: "{{ transition | float}}"
```

Comment:
These are regular service calls, why would this need documentation?

Reply:
I am adding support for transition field in https://github.com/home-assistant/core/pull/48255  and wanted to provide  working examples in the sample code, in the spirit of the rest of the documentation for this integration. 

I also found the existing sample code to be quite useful as a new user.

Reply:
I'm new to this project.  What, if any, documentation would be  a better fit in this case?

Reply:
I'm confused, the service called here, is a custom script. How would transition work if my script doesn't support it?

As for all other service calls that do support transition (e.g., `light.turn_on`), those service calls support it? Why does it need changing in the codebase and/or this example?

Reply:
It's up to the person writing the script to use the variables in a way that makes sense, if at all.  The variable is not present unless specified as part of the service call, and one can ignore the variable if if they want.  

My current use-case is a template light for my bathroom, where the "off" behaviour is a dim red nightlight, but otherwise presents itself as a regular light.  The template light lets me emulate an actual light so i can use  the colour / temperature chooser in the lovelace card, etc in ways that I couldn't with a group.  It would be nice if I could pass a transition attribute to the template light, and use that in the script, just as i would with a regular light.  

My relevant  prod configuration: https://github.com/siliconsorceress/iot/blob/3f427d3dc42d13157bd7e98618c0ac5fa74157ed/charts/home-assistant/values.yaml#L257-L302

Reply:
In the current code base the template light does not support transition at all (not even in `light.turn_on`), and  the `set_*` scripts are called from the `light.turn_on` handler.  It makes sense to propagate the new variable to all of them when it is present.